src/tests/ftest/io/ior_single_server.py

```diff
@@ -24,114 +24,116 @@
 
 import os
-import json
 
-from apricot import Test
-
-import agent_utils
-import server_utils
+from apricot import TestWithServers
+from daos_api import DaosPool
+from ior_utils import IorCommand, IorFailed
+from mpio_utils import MpioUtils
 import write_host_file
-import ior_utils
-from daos_api import DaosContext, DaosPool, DaosApiError
 
-class IorSingleServer(Test):
-    """
-    Tests IOR with Single Server config.
+
+class IorTestBase(TestWithServers):
+    """Base IOR test class.
+
     :avocado: recursive
     """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a IorTestBase object."""
+        super(IorTestBase, self).__init__(*args, **kwargs)
+        self.ior_cmd = None
+        self.processes = None
+
     def setUp(self):
-        # get paths from the build_vars generated by build
-        with open('../../../.build_vars.json') as build_file:
-            build_paths = json.load(build_file)
-        self.basepath = os.path.normpath(build_paths['PREFIX'] + "/../")
-
-        self.server_group = self.params.get("name", '/server_config/',
-                                            'daos_server')
-        self.daosctl = self.basepath + '/install/bin/daosctl'
-
-        # setup the DAOS python API
-        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
-        self.pool = None
-
-        self.hostlist_servers = self.params.get("test_servers",
-                                                '/run/hosts/test_machines/*')
-        self.hostfile_servers = (
-            write_host_file.write_host_file(self.hostlist_servers,
-                                            self.workdir))
-        print("Host file servers is: {}".format(self.hostfile_servers))
-
-        self.hostlist_clients = self.params.get("test_clients",
-                                                '/run/hosts/test_machines/*')
-        self.hostfile_clients = (
-            write_host_file.write_host_file(self.hostlist_clients,
-                                            self.workdir, None))
-        print("Host file clientsis: {}".format(self.hostfile_clients))
-
-        # set ior_flags and object class to be used by test
-        self.ior_flags = self.params.get("F", '/run/ior/iorflags/')
-        self.object_class = self.params.get("o", '/run/ior/objectclass/')
-
-        self.agent_sessions = agent_utils.run_agent(self.basepath,
-                                                    self.hostlist_servers,
-                                                    self.hostlist_clients)
-        server_utils.run_server(self.hostfile_servers, self.server_group,
-                                self.basepath)
+        """Set up each test case."""
+        # Start the servers and agents
+        super(IorTestBase, self).setUp()
+
+        # Recreate the client hostfile without slots defined
+        self.hostfile_clients = write_host_file.write_host_file(
+            self.hostlist_clients, self.workdir, None)
```

Comment:
Rather than recreating this where need be, any thoughts on setting an attribute/flag in `__init__()` that tells **apricot** whether to include slot information or not?

Reply:
Agreed, I will implement the change